pkg/applicationserver/applicationserver.go

```diff
@@ -677,7 +884,7 @@ func (as *ApplicationServer) handleDownlinkQueueInvalidated(ctx context.Context,
 			if dev == nil {
 				return nil, nil, errDeviceNotFound.WithAttributes("device_uid", unique.ID(ctx, ids))
 			}
-			if err := as.recalculateDownlinkQueue(ctx, dev, nil, invalid.Downlinks, invalid.LastFCntDown+1, link); err != nil {
+			if err := as.recalculateDownlinkQueue(ctx, dev, link, dev.Session, invalid.Downlinks, invalid.LastFCntDown+1); err != nil {
```

Comment:
This causes a `panic` if `dev.Session` is `nil`

Reply:
How can a device with no active session have its queue invalidated ?

Reply:
So what do we do with this?

Reply:
I've added a `nil` check above (before re-requesting the review you just gave).